#VRML_SIM R2021b utf8
WorldInfo {
  basicTimeStep 16
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 1 0 0 4.71238898038469
  position -0.026525516093004547 3.36643971471738 -0.0005705490799477325
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
  size 1 1
}
DEF WALL_1 Solid {
  translation 0.525 0.05 0
  scale 1.2 1.2 1.2
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.12549 0.368627 0.729412
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 0.98
      }
    }
  ]
  boundingObject USE WALL_SHAPE
}
DEF WALL_2 Solid {
  translation -0.525 0.05 0
  scale 1.2 1.2 1.2
  children [
    USE WALL_SHAPE
  ]
  name "solid(1)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_3 Solid {
  translation 0 0.05 -0.525
  rotation 0 1 0 1.5708
  scale 1.2 1.2 1.2
  children [
    USE WALL_SHAPE
  ]
  name "solid(2)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_4 Solid {
  translation 0 0.05 0.525
  rotation 0 1 0 1.5708
  scale 1.2 1.2 1.2
  children [
    USE WALL_SHAPE
  ]
  name "solid(3)"
  boundingObject USE WALL_SHAPE
}
Robot {
  translation -0.373251 0.0396265 0.36982
  rotation 0 1 0 3.13748
  children [
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 0.917647 0.145098 0.145098
        roughness 1
        metalness 0
      }
      geometry Cylinder {
        height 0.05
        radius 0.12
      }
    }
    DEF WHEEL1 HingeJoint {
      jointParameters HingeJointParameters {
        position 0.17599987176805013
        anchor 0.06 0 0
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
        PositionSensor {
          name "encoder1"
        }
      ]
      endPoint Solid {
        translation 0.06 0 0
        rotation 0.08754861785658702 -0.08754893944120663 0.9923056095347447 1.5785240680992436
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0.305882 0.898039 0.25098
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
              subdivision 24
            }
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL2 HingeJoint {
      jointParameters HingeJointParameters {
        position 0.11200009630358704
        anchor -0.06 0 0
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
        PositionSensor {
          name "encoder2"
        }
      ]
      endPoint Solid {
        translation -0.06 0 0
        rotation 0.055883113516066944 -0.05588331878658154 0.9968721745064146 1.5739327222217525
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL3 HingeJoint {
      jointParameters HingeJointParameters {
        position 0.11200009630358704
        anchor -0.06 0 0
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation 0 0 -0.06
        rotation 0.055883113516066944 -0.05588331878658154 0.9968721745064146 1.5739327222217525
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF DS_RIGHT DistanceSensor {
      translation -0.12 0 0
      rotation 0 1 0 -3.14
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.184314 0.596078 0.847059
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
      lookupTable [
        0 0 0
        1 100 0
      ]
    }
    DEF DS_LEFT DistanceSensor {
      translation 0.12 0 0
      rotation 0 -1 0 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.184314 0.596078 0.847059
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
      lookupTable [
        0 0 0
        1 100 0
      ]
    }
    DEF DS_LEFT_LEFT_FRONT DistanceSensor {
      translation 0.0866 0 0.05
      rotation 0 -1 0 0.523599
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.184314 0.596078 0.847059
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left_left_front"
      lookupTable [
        0 0 0
        1 100 0
      ]
    }
    DEF DS_LEFT_FRONT_FRONT DistanceSensor {
      translation 0.05 0 0.0866
      rotation 0 -1 0 1.0472
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.184314 0.596078 0.847059
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left_front_front"
      lookupTable [
        0 0 0
        1 100 0
      ]
    }
    DEF DS_FRONT DistanceSensor {
      translation 0 0 0.12
      rotation 0 1 0 -1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.184314 0.596078 0.847059
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_front"
      lookupTable [
        0 0 0
        1 100 0
      ]
    }
  ]
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 1
  }
  controller "odometry"
}
